File: 2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.java
class Solution {
    public long countSubarrays(int[] nums, long k) {
    //    long count = 0;
    //   int i=0;
    //   while(i < nums.length){
    //     int j = i;
    //     int sum = 0;
    //     while(j < nums.length){
    //         sum += nums[j];
    //         if(sum * (j+1-i) < k){
    //             count++;
    //         }
    //         j++;
    //     }
    //     i++;
    //   }
    //    return count;


//Sliding window
    int n = nums.length;
    long res = 0;
    long total = 0;
    for(int i=0,j=0; j<n; j++){
        total += nums[j];
        while(i <= j && total * (j+1-i) >= k){
            total -= nums[i];
            i++;
        }
        res += (j+1-i);
    }
    return res;
    }
}

File: 2394-count-subarrays-with-score-less-than-k/count-subarrays-with-score-less-than-k.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function(nums, k) {
    let n = nums.length;
    let res = 0;
    let total = 0;
    for(let i=0,j=0; j<n; j++){
        total += nums[j];
        while(i <= j && total * (j+1-i) >= k){
            total -= nums[i];
            i++;
        }
        res += (j+1-i);
    }
    return res;
};